lib/AST/ASTImporter.cpp

```diff
@@ -909,8 +909,28 @@ bool ASTNodeImporter::ImportDeclParts(NamedDecl *D, DeclContext *&DC,
                                       DeclarationName &Name, 
                                       NamedDecl *&ToD,
                                       SourceLocation &Loc) {
+  // Check if RecordDecl is in FunctionDecl parameters to avoid infinite loop.
+  // example: int struct_in_proto(struct data_t{int a;int b;} *d);
+  DeclContext *OrigDC = D->getDeclContext();
+  if(isa<RecordDecl>(D) && OrigDC->isFunctionOrMethod()) {
+    FunctionDecl *FunDecl = dyn_cast<FunctionDecl>(OrigDC);
+    if (FunDecl && FunDecl->hasBody()) {
```

Comment:
Do we need the function to have body? Is it not legal to have this construct in a prototype?

Reply:
Well, you made me think with this. It is legal indeed to have this in a prototype. This problem however is only raised when the function can be imported and in fact is juts being imported by ASTImporter. Do you think it is possible that the function import is triggered on a prototype?

Reply:
Yes, I think. Even if we do not import prototypes right now, in the future we might want to import them for merging attributes. Also, prototypes might be present in typedefs and maybe other language constructs that can be imported.

Reply:
Hmm. I'm thinking on the easy way. If the function does not have a body, we suppose that it is not because of an other ASTImporter error and we say, this is also of our consideration, so we also skip importing the function in this case. Of course after testing. WDYT?

Reply:
If the function has no body but the record ha this declaration as its declaration context, I am fine with skipping the function.